channel-jira-server/src/main/java/com/synopsys/integration/alert/channel/jira/server/model/JiraServerGlobalConfigModel.java

```diff
@@ -25,15 +25,15 @@ public JiraServerGlobalConfigModel() {
         // For serialization
     }
 
-    public JiraServerGlobalConfigModel(String id, String name, String url, String userName) {
+    public JiraServerGlobalConfigModel(String id, String name, String url, String userName, String password) {
```

Comment:
What if you had two classes related through polymorphism? One with the password, one without? Use the one with the password for _write_ operations and the one without for _read_?

Reply:
We would need an internal model that contains the password for when the job executes. Right now this model encompasses all use cases.  Maybe the obfuscate method should return a different type as you've suggested and The implementation of the Obfuscate method declares the obfuscated type that removes the password.  We would have to make some changes to the utility classes to support this.  I think we should take this up as a future enhancement.